pkg/scheduler/core/generic_scheduler.go

```diff
@@ -52,17 +51,25 @@ const (
 	minFeasibleNodesPercentageToFind = 5
 )
 
-// ErrNoNodesAvailable is used to describe the error that no nodes available to schedule pods.
-var ErrNoNodesAvailable = fmt.Errorf("no nodes available to schedule pods")
-
 // ScheduleAlgorithm is an interface implemented by things that know how to schedule pods
 // onto machines.
 // TODO: Rename this type.
 type ScheduleAlgorithm interface {
-	Schedule(context.Context, framework.Framework, *framework.CycleState, *v1.Pod) (scheduleResult ScheduleResult, err error)
+	// Schedule(context.Context, framework.Framework, *framework.CycleState, *v1.Pod) (scheduleResult ScheduleResult, err error)
 	// Extenders returns a slice of extender config. This is exposed for
 	// testing.
 	Extenders() []framework.Extender
+	Snapshot() error
```

Comment:
Why adding these functions into `ScheduleAlgorithm`? I'd expect to flatten this interface to the upper Scheduler struct.

I know you may be trying to add some interim PRs to make each PR small, but this one is organized in a strange way and doesn't like a functional completed unit.